gameListFunctions.php

```diff
@@ -56,13 +56,38 @@ function listGen(Array $objectArray) : String {
 }
 
 /**
- * This code reads the data inputted into the form and adds it to the db
+ * Checks which method to edit the database with then makes an object using the form inputs and applies the save, edit or delete function to it.
+ * If the function called is successful this function then returns you to the homepage.
+ *
+ * @param PDO $db
+ * @param String $pageCheck
+ * @return Void
  */
-function formReader($db) : Void
+
+function formReader(PDO $db, String $pageCheck) : Void
 {
-    if (isset($_POST) && in_array("", $_POST) === false) {
-        $game = new Game($_POST['name'], $_POST['genre'], $_POST['length'], $_POST['price']);
-        $game->saveGame($db);
+    if (isset($_POST) && in_array("", $_POST) === false ){
+        if($pageCheck === 'save' && array_map('is_numeric',$_POST) === ['name' => false, 'genre' => false, 'length' => true, 'price' => true]){
+            $game = new Game($_POST['name'], $_POST['genre'], $_POST['length'], $_POST['price']);
+            $test = $game->saveGame($db);
+            if($test){
```

Comment:
Do you need to repeat this if and header in each statement here? lines 73-74, 80-81 and 87-88? Think about DRY :)